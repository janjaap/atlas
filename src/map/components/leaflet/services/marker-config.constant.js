import {
  geoSearchType,
  dataSelectionType,
  detailPointType,
  straatbeeldPersonType,
  straatbeeldOrientationType
} from './icons.constant';

const markerConfig = {
  [geoSearchType]: { requestFocus: true },
  [dataSelectionType]: {},
  [detailPointType]: { requestFocus: true },
  [straatbeeldPersonType]: { requestFocus: true },
  [straatbeeldOrientationType]: { requestFocus: true }
};
export default markerConfig;
