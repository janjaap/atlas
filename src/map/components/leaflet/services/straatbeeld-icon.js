import { Icon } from 'leaflet';

import ICON_CONFIG from './icon-config.constant';

export const straatbeeldPersonIcon = () => new Icon({
  ...ICON_CONFIG.STRAATBEELD_PERSON,
  popupAnchor: null,
  shadowUrl: null,
  shadowSize: null,
  shadowAnchor: null,
  className: 'straatbeeld_person'
});

export const straatbeeldOrientationIcon = () => new Icon({
  ...ICON_CONFIG.STRAATBEELD_ORIENTATION,
  popupAnchor: null,
  shadowUrl: null,
  shadowSize: null,
  shadowAnchor: null,
  className: 'straatbeeld_orientation'
});
