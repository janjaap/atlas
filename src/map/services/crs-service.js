import L from 'leaflet';
import proj4 from 'proj4'; //eslint-disable-line
import proj4leaflet from 'proj4leaflet'; //eslint-disable-line

import CRS_CONFIG from './crs-config.constant';

function getCrs() {
  const rdSettings = CRS_CONFIG.RD;
  rdSettings.transformation.bounds =
    L.bounds.apply(null, CRS_CONFIG.RD.transformation.bounds);
  const crs = new L.Proj.CRS(
          rdSettings.code,
          rdSettings.projection,
          rdSettings.transformation
      );

  crs.distance = L.CRS.Earth.distance;
  crs.R = CRS_CONFIG.EARTH_RADIUS;

  return crs;
}

export const getRdObject = () => ({
  type: 'name',
  properties: {
    name: CRS_CONFIG.RD.code
  }
});

export default getCrs;
