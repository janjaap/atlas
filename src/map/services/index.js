import mapLayers from './map-layers/map-layers.config';
import mapPanelLayers from './map-panel-layers/map-panel-layers.config';
import mapBaseLayers from './map-base-layers/map-base-layers.config';

const TIMEOUT = 100;

export function getMapBaseLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapBaseLayers), TIMEOUT);
  });
}

export function getMapLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapLayers), TIMEOUT);
  });
}

export function getPanelLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapPanelLayers), TIMEOUT);
  });
}
