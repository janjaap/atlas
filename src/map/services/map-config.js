import BOUNDING_BOX from './bounding-box.constant';
import ENVIRONMENT, { ENVIRONMENTS } from '../../shared/environment';
import getCrs from './crs-service';

const BOUNDS = [
  BOUNDING_BOX.COORDINATES.southWest,
  BOUNDING_BOX.COORDINATES.northEast
];

const defaultConfig = {
  BASE_LAYER_OPTIONS: {
    tms: true,
    minZoom: 8,
    maxZoom: 16,
    bounds: BOUNDS
  },
  MAP_OPTIONS: {
    crs: getCrs(),
    maxBounds: BOUNDS,
    // 1.0 makes the bounds fully solid, preventing the user from dragging outside the bounds
    maxBoundsViscosity: 1.0,
    bounceAtZoomLimits: false,
    attributionControl: false,
    zoomControl: false
  },
  OVERLAY_OPTIONS: {
    identify: false,
    format: 'image/png',
    transparent: true
  },
  SCALE_OPTIONS: {
    position: 'bottomright',
    metric: true,
    imperial: false
  },
  ZOOM_OPTIONS: {
    position: 'bottomright',
    zoomInTitle: 'Inzoomen',
    zoomOutTitle: 'Uitzoomen'
  },
  DEFAULT_ZOOM_HIGHLIGHT: 14
};


const environmentConfig = {
  [ENVIRONMENTS.PRODUCTION]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['t1', 't2', 't3', 't4']
    },
    OVERLAY_ROOT: 'https://map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.PRE_PRODUCTION]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['t1', 't2', 't3', 't4']
    },
    OVERLAY_ROOT: 'https://map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.ACCEPTANCE]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['acc.t1', 'acc.t2', 'acc.t3', 'acc.t4']
    },
    OVERLAY_ROOT: 'https://acc.map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.DEVELOPMENT]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['acc.t1', 'acc.t2', 'acc.t3', 'acc.t4']
    },
    OVERLAY_ROOT: 'https://acc.map.data.amsterdam.nl/'
  }
};

const MAP_CONFIG = {
  ...defaultConfig,
  BASE_LAYER_OPTIONS: {
    ...defaultConfig.BASE_LAYER_OPTIONS,
    ...environmentConfig[ENVIRONMENT].BASE_LAYER_OPTIONS
  },
  OVERLAY_ROOT: environmentConfig[ENVIRONMENT].OVERLAY_ROOT
};

export default MAP_CONFIG;
