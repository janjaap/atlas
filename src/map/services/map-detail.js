import adressenLigplaats from '../../shared/services/adressen/adressen-ligplaats';
import adressenNummeraanduiding from '../../shared/services/adressen/adressen-nummeraanduiding';
import adressenOpenbareRuimte from '../../shared/services/adressen/adressen-openbare-ruimte';
import adressenPand from '../../shared/services/adressen/adressen-pand';
import adressenStandplaats from '../../shared/services/adressen/adressen-standplaats';
import adressenVerblijfsobject from '../../shared/services/adressen/adressen-verblijfsobject';
import bedrijfsinvesteringszone from '../../shared/services/bedrijfsinvesteringszone/bedrijfsinvesteringszone';
import explosievenGevrijwaardGebied from '../../shared/services/explosieven/explosieven-gevrijwaard-gebied';
import explosievenInslag from '../../shared/services/explosieven/explosieven-inslag';
import explosievenUitgevoerdOnderzoek from '../../shared/services/explosieven/explosieven-uitgevoerd-onderzoek';
import explosievenVerdachtGebied from '../../shared/services/explosieven/explosieven-verdacht-gebied';
import gebiedenBouwblok from '../../shared/services/gebieden/gebieden-bouwblok';
import gebiedenBuurt from '../../shared/services/gebieden/gebieden-buurt';
import gebiedenGebiedsgerichtWerken from '../../shared/services/gebieden/gebieden-gebiedsgericht-werken';
import gebiedenGrootstedelijk from '../../shared/services/gebieden/gebieden-grootstedelijk';
import gebiedenStadsdeel from '../../shared/services/gebieden/gebieden-stadsdeel';
import gebiedenUnesco from '../../shared/services/gebieden/gebieden-unesco';
import gebiedenWijk from '../../shared/services/gebieden/gebieden-wijk';
import grondexploitatie from '../../shared/services/grondexploitatie/grondexploitatie';
import kadastraalObject from '../../shared/services/kadastraal-object/kadastraal-object';
import meetbout from '../../shared/services/meetbout/meetbout';
import monument from '../../shared/services/monument/monument';
import napPeilmerk from '../../shared/services/nap-peilmerk/nap-peilmerk';
import vestiging from '../../shared/services/vestiging/vestiging';

export const maxDisplayValuesPerProperty = 5;

// NOTE: beware, if FETCH_DETAIL action payload includes endpointTypes string
// then magic happens in '/src/reducers/details.js'
export const endpointTypes = {
  adressenLigplaats: 'bag/ligplaats/',
  adressenNummeraanduiding: 'bag/nummeraanduiding/',
  adressenOpenbareRuimte: 'bag/openbareruimte/',
  adressenPand: 'bag/pand/',
  adressenStandplaats: 'bag/standplaats/',
  adressenVerblijfsobject: 'bag/verblijfsobject/',
  bedrijfsinvesteringszone: 'vsd/biz/',
  explosievenGevrijwaardGebied: 'milieuthemas/explosieven/gevrijwaardgebied/',
  explosievenInslag: 'milieuthemas/explosieven/inslagen/',
  explosievenUitgevoerdOnderzoek: 'milieuthemas/explosieven/uitgevoerdonderzoek/',
  explosievenVerdachtGebied: 'milieuthemas/explosieven/verdachtgebied/',
  gebiedenBouwblok: 'gebieden/bouwblok/',
  gebiedenBuurt: 'gebieden/buurt/',
  gebiedenGebiedsgerichtWerken: 'gebieden/gebiedsgerichtwerken/',
  gebiedenGrootstedelijk: 'gebieden/grootstedelijkgebied/',
  gebiedenStadsdeel: 'gebieden/stadsdeel/',
  gebiedenUnesco: 'gebieden/unesco/',
  gebiedenWijk: 'gebieden/buurtcombinatie/',
  grondexploitatie: 'grondexploitatie/project/',
  kadastraalObject: 'brk/object/',
  meetbout: 'meetbouten/meetbout/',
  monument: 'monumenten/monumenten/',
  napPeilmerk: 'nap/peilmerk/',
  vestiging: 'handelsregister/vestiging/'
};

const servicesByEndpointType = {
  [endpointTypes.adressenLigplaats]: { fetch: adressenLigplaats },
  [endpointTypes.adressenNummeraanduiding]: { fetch: adressenNummeraanduiding },
  [endpointTypes.adressenOpenbareRuimte]: { fetch: adressenOpenbareRuimte },
  [endpointTypes.adressenPand]: { fetch: adressenPand },
  [endpointTypes.adressenStandplaats]: { fetch: adressenStandplaats },
  [endpointTypes.adressenVerblijfsobject]: { fetch: adressenVerblijfsobject },
  [endpointTypes.bedrijfsinvesteringszone]: { fetch: bedrijfsinvesteringszone },
  [endpointTypes.explosievenGevrijwaardGebied]: { fetch: explosievenGevrijwaardGebied },
  [endpointTypes.explosievenInslag]: { fetch: explosievenInslag },
  [endpointTypes.explosievenUitgevoerdOnderzoek]: { fetch: explosievenUitgevoerdOnderzoek },
  [endpointTypes.explosievenVerdachtGebied]: { fetch: explosievenVerdachtGebied },
  [endpointTypes.gebiedenBouwblok]: { fetch: gebiedenBouwblok },
  [endpointTypes.gebiedenBuurt]: { fetch: gebiedenBuurt },
  [endpointTypes.gebiedenGebiedsgerichtWerken]: { fetch: gebiedenGebiedsgerichtWerken },
  [endpointTypes.gebiedenGrootstedelijk]: { fetch: gebiedenGrootstedelijk },
  [endpointTypes.gebiedenStadsdeel]: { fetch: gebiedenStadsdeel },
  [endpointTypes.gebiedenUnesco]: { fetch: gebiedenUnesco },
  [endpointTypes.gebiedenWijk]: { fetch: gebiedenWijk },
  [endpointTypes.grondexploitatie]: { fetch: grondexploitatie, authScope: 'GREX/R' },
  [endpointTypes.kadastraalObject]: { fetch: kadastraalObject },
  [endpointTypes.meetbout]: { fetch: meetbout },
  [endpointTypes.monument]: { fetch: monument },
  [endpointTypes.napPeilmerk]: { fetch: napPeilmerk },
  [endpointTypes.vestiging]: { fetch: vestiging, authScope: 'HR/R' }
};

export default function fetchDetail(endpoint, user) {
  const endpointType = Object.keys(servicesByEndpointType).find((type) => endpoint.includes(type));
  const endpointConfig = endpointType && servicesByEndpointType[endpointType];
  const fetchFn = endpointConfig && endpointConfig.fetch;
  const authScope = endpointConfig && endpointConfig.authScope;
  return fetchFn && (!authScope || user.scopes.includes(authScope)) &&
    fetchFn(endpoint);
}

window.mapPreviewPanelDetailEndpointTypes = endpointTypes;
