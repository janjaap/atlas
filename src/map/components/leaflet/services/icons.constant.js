import searchIcon from './search-icon';
import { straatbeeldPersonIcon, straatbeeldOrientationIcon } from './straatbeeld-icon';
import detailIcon from './detail-icon';
import dataSelectionIcon from './data-selection-icon';

export const geoSearchType = 'geoSearchType';
export const dataSelectionType = 'dataSelectionType';
export const detailPointType = 'detailPointType';
export const straatbeeldPersonType = 'straatbeeldPersonType';
export const straatbeeldOrientationType = 'straatbeeldOrientationType';

const icons = {
  [geoSearchType]: searchIcon,
  [dataSelectionType]: dataSelectionIcon,
  [detailPointType]: detailIcon,
  [straatbeeldPersonType]: straatbeeldPersonIcon,
  [straatbeeldOrientationType]: straatbeeldOrientationIcon
};
export default icons;
