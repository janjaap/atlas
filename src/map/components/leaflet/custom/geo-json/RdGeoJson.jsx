import { GeoJSON } from 'react-leaflet';
import L from 'leaflet';
import proj4 from 'proj4'; //eslint-disable-line
import proj4leaflet from 'proj4leaflet'; //eslint-disable-line

import { getRdObject } from '../../../../services/crs-service';
import ICON_CONFIG from '../../services/icon-config.constant';

class RdGeoJson extends GeoJSON {
  createLeafletElement() {
    const { data, style } = this.props;
    const geometry = {
      ...data.geometry,
      crs: getRdObject()
    };
    const customStyle = {
      color: 'red',
      fillColor: 'red',
      weight: 2,
      opacity: 1.6,
      fillOpacity: 0.2,
      ...style
    };
    return L.Proj.geoJson(geometry, {
      style: /* istanbul ignore next */ () => customStyle,
      pointToLayer: /* istanbul ignore next */ (feature, latLng) => {
        const icon = L.icon(ICON_CONFIG.DETAIL);
        const rotationAngle = data.orientation || 0;

        return L.marker(latLng, {
          icon,
          rotationAngle,
          alt: 'Marker op de kaart'
        });
      }
    });
  }

  componentWillUnmount() {
    super.componentWillUnmount();
    this.layerContainer.removeLayer(this.leafletElement);
  }
}

export default RdGeoJson;
