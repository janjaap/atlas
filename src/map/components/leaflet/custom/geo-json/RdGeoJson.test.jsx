import React from 'react';
import { shallow } from 'enzyme';
import leafletModule from 'leaflet';
import RdGeoJson from './RdGeoJson';
import '../../../../services/crs-service';

leafletModule.Proj.geoJson = jest.fn();

function shallowRenderComponent() {
  return shallow(
    <RdGeoJson
      data={{
        geometry: {
          type: 'MultiPolygon',
          coordinates: [
            [
              [
                [
                  121354.82,
                  487493.672
                ],
                [
                  121355.332,
                  487498.93
                ],
                [
                  121349.41,
                  487499.506
                ],
                [
                  121348.903,
                  487494.304
                ],
                [
                  121354.82,
                  487493.672
                ]
              ]
            ]
          ]
        },
        label: 'ASD04 F 04358 G 0000'
      }}
      key="https://acc.api.data.amsterdam.nl/brk/object/NL.KAD.OnroerendeZaak.11450435870000/"
      ref={jest.fn()}
    />, {
      context: {
        layerContainer: {
          removeLayer: jest.fn()
        }
      },
      disableLifecycleMethods: true
    }
  );
}

describe('RdGeoJson component', () => {
  it('should call geoJson with expected arguments', () => {
    shallowRenderComponent();
    expect(leafletModule.Proj.geoJson.mock.calls).toMatchSnapshot();
  });

  it('should call super.componentWillUnmount', () => {
    const component = shallowRenderComponent();
    component.instance().componentWillUnmount();
    expect(component.instance().layerContainer.removeLayer).toHaveBeenCalled();
  });
});
