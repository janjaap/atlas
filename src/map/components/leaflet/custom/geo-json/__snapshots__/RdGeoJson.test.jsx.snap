// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RdGeoJson component should call geoJson with expected arguments 1`] = `
Array [
  Array [
    Object {
      "coordinates": Array [
        Array [
          Array [
            Array [
              121354.82,
              487493.672,
            ],
            Array [
              121355.332,
              487498.93,
            ],
            Array [
              121349.41,
              487499.506,
            ],
            Array [
              121348.903,
              487494.304,
            ],
            Array [
              121354.82,
              487493.672,
            ],
          ],
        ],
      ],
      "crs": Object {
        "properties": Object {
          "name": "EPSG:28992",
        },
        "type": "name",
      },
      "type": "MultiPolygon",
    },
    Object {
      "pointToLayer": [Function],
      "style": [Function],
    },
  ],
]
`;
